### PR TITLE
Solution to #109

### PR DESCRIPTION
**Pull request by FunFunFine**
Made it possible to summon implicit instances from context

### CODE REVIEW

File: core/src/main/scala/tofu/package.scala [hunk 1]
```diff
@@ -2,13 +2,25 @@ import cats.effect.Bracket
 import cats.{ApplicativeError, MonadError}
 
 package object tofu {
-  type HasContext[F[_], C]          = Context[F] { type Ctx    = C }
-  type HasLocal[F[_], C]            = Local[F] { type Ctx      = C }
+  type HasContext[F[_], C] = Context[F] { type Ctx = C }
+  object HasContext {
```
Comment (Odomontois) [inline core/src/main/scala/tofu/package.scala]: I fill those belong to Errors.scala

Comment (Z1kkurat) [reply core/src/main/scala/tofu/package.scala]: @odomontois Why do you think summoning instances belongs to Errors file?

Comment (Odomontois) [reply core/src/main/scala/tofu/package.scala]: package objects are nasty things in terms of refactoring, bytecode, binary compatibility
I'd like to have them as small as possible, so all the things, that are defineable in the package itself, should be defined in the package itself, I feel

**Comment by FunFunFine** [comment]
finally moved summoners to Errors.scala as @Odomontois said

**Comment by Z1kkurat** [comment]
@FunFunFine can you please format the files? Seems like our moder formatter doesn't quite like current formatting: https://travis-ci.com/TinkoffCreditSystems/tofu/jobs/282971940#L269

**Comment by Z1kkurat** [comment]
LGTM. @Odomontois if you don't have any objections I will merge this

**Pull request merged**